api/views/auth.py

```diff
@@ -11,45 +11,38 @@
 
 mod = Blueprint('auth', __name__)
 
-SIGNUP_URL = '/signup'
-LOGIN_URL = '/login'
-LOGOUT_URL = '/logout'
+SIGNUP_URL = '/auth/signup'
+LOGIN_URL = '/auth/login'
+LOGOUT_URL = '/auth/logout'
+TEST_URL = '/auth/test-token-required'
 
 
 @app.route(SIGNUP_URL, methods=['POST'])
 def create_account():
     request_json = request.get_json()
 
-    keys = set(['username', 'password', 'accepts_email', 'email',
-            'custom_email_subject', 'custom_email_body'])
-    missing_keys = keys - set(request_json.keys())
-    if missing_keys:
+    required_fields = ['username', 'password']
+    missing_fields = [field for field in required_fields if request_json.get(field) is None]
+    if len(missing_fields):
         return create_response(status=400,
-                message='request json is missing the following keys: {}'\
-                .format(missing_keys))
+                message='Request json is missing the following fields: {}' \
+                .format(', '.join(missing_fields)))
 
-    username = request_json['username']
-    password = request_json['password']
-    accepts_email = request_json['accepts_email']
-    email = request_json['email']
-    custom_email_subject = request_json['custom_email_subject']
-    custom_email_body = request_json['custom_email_body']
+    username = request_json.get('username')
+    password = request_json.get('password')
 
     existing_user = User.query.filter_by(username=username).first()
-    if existing_user:
+    if existing_user is not None:
         return create_response(status=400,
-                message='username "{}" is already taken'.format(username))
+                message='Username "{}" is already taken'.format(username))
 
     password_fail_message = auth_utils.validate_password_requirements(password)
     if password_fail_message:
         return create_response(status=400, message=password_fail_message)
 
     salt = auth_utils.generate_salt()
     hashed_password = auth_utils.hash_password(password, salt)
-    new_user = User(username=username, accepts_email=accepts_email, email=email,
-                    custom_email_subject=custom_email_subject,
-                    custom_email_body=custom_email_body,
-                    salt=salt, pw_hash=hashed_password)
+    new_user = User(username=username, salt=salt, pw_hash=hashed_password)

```

Comment:
would it make sense to use `row_constructor` here or nah?

Reply:
you would have to make the new dictionary containing all of those things rather than use something that already exists, so I think it's fine to not use `row_constructor` here